opm/simulators/linalg/twolevelmethodcpr.hh

```diff
@@ -228,6 +227,30 @@ public:
   }
 
 private:
+  template <typename RN, typename PI, typename PG, typename VM, typename AM>
+  auto coarsen(RN& renumberer,
+               PI pinfo,
+               PG pg,
+               VM vm,
+               std::shared_ptr<AM>* aggregatesMap,
```

Comment:
I assume there is a reason for taking a pointer to a `shared_ptr<>` as a parameter&ndash;e.g., because `renumberer.coarsen()` wants to participate in lifetime management of the underlying reason?  Otherwise, the typical parameter type would be (possibly cv qualified) `AM*` or `AM&` in this case.

Am I missing something?

Reply:
It prevents one additional copy of the shared_ptr, which would otherwise be introduced just because of this. renumber.coarsen takes a shared_ptr for whatever reason (this interface is ancient :flushed: ).